databroker/core.py

```diff
@@ -833,3 +833,90 @@ def __iter__(self):
                 yield doc
             else:
                 self.__stash_values(name, doc)
+
+
+class HeaderSourceShim(object):
```

Comment:
but I think this is how it should work, the object that wraps the start/stop collections should know these semantics.  As it is now we are insisting that `find_run_start` implement it exactly the way that makes sense for the mongo derived collections.